yb-voyager/cmd/exportData.go

```diff
@@ -1780,12 +1781,29 @@ func createUpdateExportedRowCountEventList(tableNames []string) []*cp.UpdateExpo
 	return result
 }
 
-func saveTableToUniqueKeyColumnsMapInMetaDB(tableList []sqlname.NameTuple) {
+func saveTableToUniqueKeyColumnsMapInMetaDB(tableList []sqlname.NameTuple, leafPartitions *utils.StructMap[sqlname.NameTuple, []sqlname.NameTuple]) {
 	res, err := source.DB().GetTableToUniqueKeyColumnsMap(tableList)
 	if err != nil {
 		utils.ErrExit("get table to unique key columns map: %w", err)
 	}
 
+	//Adding all the leaf partitions unique key columns to the root table unique key columns since in the importer all the events only have the root table name
+	leafPartitions.IterKV(func(key sqlname.NameTuple, value []sqlname.NameTuple) (bool, error) {
+		keyTbl := key.AsQualifiedCatalogName()
+		for _, leaf := range value {
+			keyLeafTbl := leaf.AsQualifiedCatalogName()
+			leafColumns, ok := res[keyLeafTbl]
+			if !ok {
+				continue
+			}
+			//Do not add leaf table key in the map since this config will be read by importer
+			delete(res, keyLeafTbl)
+			res[keyTbl] = append(res[keyTbl], leafColumns...)
```

Comment:
what happens if all the leaf tables have the same set of unique columns. 
They will get added to the list multiple times, with duplicates?